tls/s2n_early_data.c

```diff
@@ -31,6 +31,9 @@ const s2n_early_data_state valid_previous_states[] = {
 S2N_RESULT s2n_connection_set_early_data_state(struct s2n_connection *conn, s2n_early_data_state next_state)
 {
     ENSURE_REF(conn);
+    if (conn->early_data_state == next_state) {
+        return S2N_RESULT_OK;
+    }
```

Comment:
No, because that's not the only possible no-op. The same statement can be made for any state. Do you think it's unclear that this means no-ops are allowed?